### PR TITLE
Updated CondStartsEndsWith to support multiple strings

### PR DESCRIPTION
**Pull request by DeltaRays**
---
**Target Minecraft Versions:** any 
**Requirements:** none 
**Related Issues:** #3312

### CODE REVIEW

File: src/main/java/ch/njol/skript/conditions/CondStartsEndsWith.java [hunk 1]
```diff
@@ -34,15 +34,15 @@
 
 @Name("Starts/Ends With")
 @Description("Checks if a text starts or ends with another.")
-@Examples({"if the argument starts with \"test\":",
-		"	send \"Stop!\""})
-@Since("2.2-dev36")
+@Examples({"if the argument starts with \"test\" or \"debug\":",
+	"	send \"Stop!\""})
```
Comment (APickledWalrus) [inline src/main/java/ch/njol/skript/conditions/CondStartsEndsWith.java:38]: Can you replace the tab here with `\t`? Since you're already editing this expression ;D

File: src/main/java/ch/njol/skript/conditions/CondStartsEndsWith.java [hunk 2]
```diff
@@ -63,15 +63,38 @@ public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelaye
...
 	@Override
 	public boolean check(Event e) {
-		String affix = this.affix.getSingle(e);
-		
-		if (affix == null) {
-			return false;
-		}
-		
+		String[] affixes = this.affix.getAll(e);
+		if (affixes.length < 1) return false;
```
Comment (APickledWalrus) [inline src/main/java/ch/njol/skript/conditions/CondStartsEndsWith.java:67]: Nitpick but I think
```java
return false;
```
should be on its own line.

File: src/test/skript/tests/syntaxes/conditions/CondStartsEndsWith.sk [hunk 3]
```diff
@@ -0,0 +1,9 @@
+test "starts with condition":
+	assert "hey" starts with "he" or "hello" with "hey doesn't start with he or hello"
+	assert "hey" starts with "hey" and "h" with "hey doesn't start with hey and h"
+	assert "hey" doesn't start with "hey" and "e" with "hey starts with hey and e"
+
+test "ends with condition":
+	assert "hey" ends with "hey" or "hello" with "hey doesn't end with hey or hello"
+	assert "hey" ends with "hey" and "y" with "hey doesn't end with hey and y"
+	assert "hey" doesn't end with "hey" and "e" with "hey ends with hey and e"
```
Comment (APickledWalrus) [inline src/test/skript/tests/syntaxes/conditions/CondStartsEndsWith.sk:9]: can you add a newline at the end of the file

File: src/main/java/ch/njol/skript/conditions/CondStartsEndsWith.java [hunk 4]
```diff
@@ -63,15 +63,38 @@ public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelaye
...
+						if (!string.startsWith(str))
+							return false;
+					}
+					return true;
+				} else {
+					for (String str : affixes) {
+						if (string.startsWith((str)))
+							return true;
+					}
+				}
```
Comment (APickledWalrus) [inline src/main/java/ch/njol/skript/conditions/CondStartsEndsWith.java:94]: I think this will improve readability, but it's open to discussion.
```suggestion
				} else { // Using 'starts with'
					if (this.affix.getAnd()) {
						for (String str : affixes) {
							if (!string.startsWith(str))
								return false;
						}
						return true;
					} else {
						for (String str : affixes) {
							if (string.startsWith((str)))
								return true;
						}
					}
				}
```

**Review by APickledWalrus — CHANGES_REQUESTED** [review]
Just a few little things. Good PR - this will be really useful.

File: src/main/java/ch/njol/skript/conditions/CondStartsEndsWith.java (hunk 4 above)
Comment (DeltaRays) [reply src/main/java/ch/njol/skript/conditions/CondStartsEndsWith.java:95]: Yeah

**Review by APickledWalrus — APPROVED** [review]
LGTM

File: src/main/java/ch/njol/skript/conditions/CondStartsEndsWith.java [hunk 5]
```diff
@@ -1,18 +1,18 @@
...
- *  GNU General Public License for more details.
+ * Skript is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
  *
- *  You should have received a copy of the GNU General Public License
- *  along with Skript.  If not, see <http://www.gnu.org/licenses/>.
+ * You should have received a copy of the GNU General Public License
+ * along with Skript.  If not, see <http://www.gnu.org/licenses/>.
```
Comment (ShaneBeee) [inline src/main/java/ch/njol/skript/conditions/CondStartsEndsWith.java:15]: As stated in another PR, please revert this back to the default license header.
Line changes which do not affect the PR shouldn't be part of the PR.

you can find a copy of the default header here:
https://github.com/SkriptLang/Skript/blob/master/licenseheader.txt

**Review by ShaneBeee — CHANGES_REQUESTED** [review]
just a small change.

File: src/main/java/ch/njol/skript/conditions/CondStartsEndsWith.java (hunk 5 above)
Comment (DeltaRays) [reply src/main/java/ch/njol/skript/conditions/CondStartsEndsWith.java:15]: Thanks, I’ll do it when I come home

**Review by APickledWalrus — APPROVED** [review]
LGTM

**Review by ShaneBeee — APPROVED** [review]

**Comment by ShaneBeee** [comment]
Thank you for your contribution
